docs/en/sql-reference/functions/math-functions.md

```diff
@@ -549,3 +549,33 @@ Result:
 │ 3.141592653589793 │
 └───────────────────┘
 ```
+
+
+## factorial(n)
+
+Computes the factorial of an integer value. It works with any native integer type including UInt(8|16|32|64) and Int(8|16|32|64). The return type is Int64.
+
+The factorial of 0 is 1. Likewise, the factorial() function returns 1 for any negative value. The maximum positive value for the input argument is 20; a value of 21 or greater overflows the range for Int64 and will causes an throw exception.
```

Comment:
Consider to rename `will causes an throw` into `will throw exception`, `will cause exception throw`.

Reply:
I also `is 20; a` maybe better to rename into `is 20, a`.